### DISCUSSION TITLE
Imitating tail -f

### DISCUSSION THREAD

**Discussion by jsatchell** [gyscos/cursive#746] [Q&A]
I would like to have a component that tracks the content of an external log file, so the user can see progress during a long running external process. Polled updates would be fine, or I could use a reader thread and send lines down a channel to the UI thread as they become available. Is there an example that shows how to do something like this? I found the progressbar, but that is a special case set up for updates over a channel. Is there a simple pattern for adding content to view and flushing it to screen?

**Comment by gyscos** [comment]
Hi, and thanks for the question!

The main ways to update a view:
* Get a "callback sync" (the sender half of a channel) to send UI callbacks that can alter the view tree, for example adding or editing views.
* Use a shared structure so "writes" on the producer side can be seen from the view. This can also be a channel (for example the [logs example](https://github.com/gyscos/cursive/blob/main/cursive/examples/logs.rs) uses a channel of `String` to send individual log entries), or it could be a simple `Arc<Mutex<Vec<String>>>` or similar structure. Each has its own tradeoffs regarding locking and re-allocations.
